tablecodec/tablecodec.go

```
@@ -204,11 +204,9 @@ func flatten(data types.Datum, loc *time.Location) (types.Datum, error) {
 	case types.KindMysqlSet:
 		data.SetUint64(data.GetMysqlSet().Value)
 		return data, nil
-	case types.KindMysqlBit:
-		data.SetUint64(data.GetMysqlBit().Value)
-		return data, nil
-	case types.KindMysqlHex:
-		data.SetInt64(data.GetMysqlHex().Value)
+	case types.KindBinaryLiteral, types.KindMysqlBit:
+		val, _ := data.GetBinaryLiteral().ToInt()
```

error is not handled.